app/src/main/java/ch/epfl/skysync/dataModels/userModels/User.kt

```diff
@@ -0,0 +1,19 @@
+package ch.epfl.skysync.dataModels.userModels
+
+import ch.epfl.skysync.dataModels.calendarModels.AvailabilityCalendar
+import ch.epfl.skysync.dataModels.calendarModels.FlightCalendar
+import ch.epfl.skysync.dataModels.flightModels.RoleType
+
+interface  User {
+    val firstname: String
+    val lastname: String
+    val userId: String
+    val availabilities: AvailabilityCalendar
+    val assignedFlights: FlightCalendar
+
+    fun addRoleType(roleType: RoleType): User
+
+    fun hasRoleType(roleType: RoleType): Boolean
```

Comment:
Why not have `roleTypes` defined in `User`? That way `hasRoleType`, `addRoleType` could be define in `User` too.

Reply:
Right now you just have duplicated code, as they all have a `roleTypes` attribute anyway, as for the restriction on the roles that can be added, you can just override the default method.